dahee/services/userService.js

```diff
@@ -10,9 +13,34 @@ const signUp = async (name, email, password, profileImage) => {
     err.statusCode = 400;
     throw err;
   }
-  return userDao.createUser(name, email, password, profileImage);
+
+  const saltRounds = 12;
+  const hashedPassword = await bcrypt.hash(password, saltRounds);
+  return userDao.createUser(name, email, hashedPassword, profileImage);
+};
+
+const signIn = async (email, password) => {
+  try {
+    const user = await userDao.verifyUserByEmail(email);
+    const authResult = await bcrypt.compare(password, user.password);
+
+    if (!authResult) {
+      const error = new Error('INVALID_USER');
+      error.statusCode = 401;
+      throw error;
+    }
```

Comment:
1. email이 존재하지 않은 경우 user에는 빈 값이 들어갈 것입니다. 따라서 user.email을 비교하는 것이 아니고 user가 존재하는지 자체를 판단해야 됩니다.
2. user가 존재 하지 않으면 비밀번호를 확인 할 이유가 없기 때분에 로직순서는 아래와 같아야 합니다.
4. 이메일이 없을 때와 비밀번호가 틀렸을 때 에러 메시지는 통일 시켜주는 것이 좋습니다.
```suggestion
  
  if (!user) {
    const err = new Error('INVALID_USER');
    err.statusCode = 401;
    throw err;
  }
  
  const isMatched = await bcrypt.compare(password, user.password);
  
  if (!isMatched) {
    const err = new Error('NOT_USER');
    err.statusCode = 401;
    throw err;
  }
```